test/common/config/delta_subscription_impl_test.cc

```diff
@@ -0,0 +1,111 @@
+#include "common/config/delta_subscription_impl.h"
+
+#include "test/mocks/config/mocks.h"
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/grpc/mocks.h"
+#include "test/mocks/local_info/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+#include "test/mocks/stats/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::InSequence;
+using testing::NiceMock;
+using testing::Return;
+
+namespace Envoy {
+namespace Config {
+namespace {
+
+typedef DeltaSubscriptionImpl<envoy::api::v2::ClusterLoadAssignment> DeltaEdsSubscriptionImpl;
+
+class DeltaSubscriptionImplTest : public testing::Test {
```

Comment:
I would advocate to factor this into harness/test as done for the other config tests. That way, we can reuse common tests (and write specialized ones where needed).

Reply:
I created delta subscription test harness and attached it to common subscription tests in subscription_impl_test.cc.